src/policy/mallocspace.rs

```diff
@@ -0,0 +1,123 @@
+pub struct MallocSpace<VM: VMBinding> {
+    phantom: PhantomData<VM>,
+    pr: MonotonePageResource<VM>,
+}
+
+impl<VM: VMBinding> SFT for MallocSpace<VM> {
+    fn name(&self) -> &str {
+        "MallocSpace"
+    }
+
+    fn is_live(&self, _object: ObjectReference) -> bool {
+        unimplemented!();
+    }
+    fn is_movable(&self) -> bool {
+        unimplemented!();
+    }
+    #[cfg(feature = "sanity")]
+    fn is_sane(&self) -> bool {
+        unimplemented!();
+    }
+    fn initialize_header(&self, _object: ObjectReference, _alloc: bool) {}
+}
+
+impl<VM: VMBinding> Space<VM> for MallocSpace<VM> {
+    fn as_space(&self) -> &dyn Space<VM> {
+        unimplemented!();
+    }
+    fn as_sft(&self) -> &(dyn SFT + Sync + 'static) {
+        unimplemented!();
+    }
+    fn get_page_resource(&self) -> &dyn PageResource<VM> {
+        &self.pr
+    }
+    fn common(&self) -> &CommonSpace<VM> {
+        unimplemented!();
+    }
+    unsafe fn unsafe_common_mut(&self) -> &mut CommonSpace<VM> {
+        unimplemented!();
+    }
+
+    fn init(&mut self, _vm_map: &'static VMMap) {
+        unimplemented!();
+    }
+
+    fn release_multiple_pages(&mut self, _start: Address) {
+        unimplemented!();
+    }
+
+    fn in_space(&self, object: ObjectReference) -> bool {
+        let address = object.to_address();
+        self.address_in_space(address)
+    }
+
+    fn address_in_space(&self, start: Address) -> bool {
+        meta_space_mapped(start) && load_atomic(ALLOC_METADATA_SPEC, start) == 1
+    }
+
+    fn get_name(&self) -> &'static str {
+        "MallocSpace"
+    }
+
+    fn reserved_pages(&self) -> usize {
+        self.pr.common().get_reserved()
+    }
+
+    unsafe fn release_all_chunks(&self) {
+        let mut released_chunks = HashSet::new();
+        for chunk_start in &*ACTIVE_CHUNKS.read().unwrap() {
+            let mut chunk_is_empty = true;
+            let mut address = *chunk_start;
+            let chunk_end = chunk_start.add(BYTES_IN_CHUNK);
+            while address.as_usize() < chunk_end.as_usize() {
+                if load_atomic(ALLOC_METADATA_SPEC, address) == 1 {
+                    if !is_marked(address) {
+                        let ptr = address.to_mut_ptr();
+                        free(ptr);
+                        unset_alloc_bit(address);
+                    } else {
+                        unset_mark_bit(address);
+                        chunk_is_empty = false;
+                    }
+                }
+                address = address.add(VM::MAX_ALIGNMENT);
+            }
+            if chunk_is_empty {
+                released_chunks.insert(chunk_start.as_usize());
+            }
+        }
+        self.pr.common().release_reserved(PAGES_IN_CHUNK * released_chunks.len());
+
+        ACTIVE_CHUNKS
+            .write()
+            .unwrap()
+            .retain(|c| !released_chunks.contains(&c.as_usize()));
+    }
+}
+
+impl<VM: VMBinding> MallocSpace<VM> {
```

Comment:
We should have a `trace_object()` function about how to trace object in this policy.